sdk/src/lib.rs

```diff
@@ -215,11 +328,15 @@ pub use solana_sdk_macro::declare_deprecated_id;
 pub use solana_sdk_macro::declare_id;
 /// Convenience macro to define multiple static public keys.
 pub use solana_sdk_macro::pubkeys;
-#[deprecated(since = "2.2.0", note = "Use `solana-secp256k1-program` crate instead")]
+#[deprecated(since = "2.3.0", note = "Use `solana-secp256k1-program` crate instead")]
 #[cfg(feature = "full")]
-pub use solana_secp256k1_program as secp256k1_instruction;
-#[deprecated(since = "2.1.0", note = "Use `solana-secp256k1-recover` crate instead")]
-pub use solana_secp256k1_recover as secp256k1_recover;
+pub mod secp256k1_instruction {
+    pub use solana_secp256k1_program::*;
+}
+#[deprecated(since = "2.3.0", note = "Use `solana-secp256k1-recover` crate instead")]
+pub mod secp256k1_recover {
+    pub use solana_secp256k1_recover::*;
+}
```

Comment:
these would also be more consistent as `solana_*_interface`?

Reply:
`solana-secp256k1-program` can do the same as `solana-ed25519-program`.

The recover crate is a bit strange, since it's just one syscall and a fallback implementation for non-solana targets. I'd prefer to keep it as is since it has some crypto deps.

Reply:
imo we need to work toward a consistent interface here. absurd that there's a Precompile trait and that there's like zero consistency between the implementations

Reply:
Yep, agreed big-time